### ISSUE TITLE
the pictures and examples do not match

### ISSUE THREAD

**Issue by Zzj-pcl** [AbangLZU/VoxelNetRos#6]
![Screenshot from 2019-04-29 11-51-17](https://user-images.githubusercontent.com/40758881/56875746-51f15880-6a75-11e9-9619-9611fc5b2ab5.png)

Why do I run the results of the pictures and examples do not match?

**Comment by Zzj-pcl** [comment]
I get it

**Comment by scarlettliu644** [comment]
Hi, may I ask how did u solve it?

> I get it

**Comment by Zzj-pcl** [comment]
> Hi, may I ask how did u solve it?
> 
> > I get it



> Hi, may I ask how did u solve it?
> 
> > I get it

What is your problem?

**Comment by Zzj-pcl** [comment]
What is your problem?

**Comment by Zzj-pcl** [comment]
> Hi, may I ask how did u solve it?
> 
> > I get it

What is your problem?

**Comment by scarlettliu644** [comment]
I have the same question as you posted, "the pictures and examples do not match".  I asked "how did u solve it" since you posted your second comment of " I get it". I thought you solved your own question so may I ask how did you solve it? 

Mine results are not good too. The detected box does not match the modified points.

**Comment by Zzj-pcl** [comment]
> I have the same question as you posted, "the pictures and examples do not match". I asked "how did u solve it" since you posted your second comment of " I get it". I thought you solved your own question so may I ask how did you solve it?
> 
> Mine results are not good too. The detected box does not match the modified points.

I tested the vehicle with velodyne-16 and vehicle detection using tensorflow-gpu as shown.
Initially, the detection box should not be correct because the parameters were not adjusted in rviz.
Later, by adjusting the parameters in BoundingBox and BoundingBox Array, I made Box detect correctly.

![image](https://user-images.githubusercontent.com/40758881/58378842-599d1200-7fcd-11e9-970e-25f542507101.png)
As shown in the figure， tick only edge to OK.

**Comment by scarlettliu644** [comment]
Thanks for your quick reply.  

My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly.  Still I have a couple of questions if you don't mind to exchange ideas:

1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours? 

2. How's your detection accuracy? 

Appreciate your sharing. 

Cheers.

**Comment by Zzj-pcl** [comment]
> Thanks for your quick reply.
> 
> My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly. Still I have a couple of questions if you don't mind to exchange ideas:
> 
> 1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours?
> 2. How's your detection accuracy?
> 
> Appreciate your sharing.
> 
> Cheers.



> Thanks for your quick reply.
> 
> My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly. Still I have a couple of questions if you don't mind to exchange ideas:
> 
> 1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours?
> 2. How's your detection accuracy?
> 
> Appreciate your sharing.
> 
> Cheers.

**Comment by Zzj-pcl** [comment]
> > Thanks for your quick reply.
> > My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly. Still I have a couple of questions if you don't mind to exchange ideas:
> > 
> > 1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours?
> > 2. How's your detection accuracy?
> > 
> > Appreciate your sharing.
> > Cheers.
> 
> > Thanks for your quick reply.
> > My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly. Still I have a couple of questions if you don't mind to exchange ideas:
> > 
> > 1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours?
> > 2. How's your detection accuracy?
> > 
> > Appreciate your sharing.
> > Cheers.

**Comment by Zzj-pcl** [comment]
> Thanks for your quick reply.
> 
> My understanding is that the BoundingBox has no effect on the final visualization. I did adjust BoundingBoxArray and overlay it on the ModifiedPoints and the results seem matched. However, loads of false positive on my VLP16 datasets. The detected boxes are far off from the object it is supposed to detect. Hence I commented on our topic by asking how did you manage the detection correctly. Still I have a couple of questions if you don't mind to exchange ideas:
> 
> 1. The detection speed on my PC is quite slow (nothing to do with the hardware since my PC is quite good). How's yours?
> 2. How's your detection accuracy?
> 
> Appreciate your sharing.
> 
> Cheers.

Yes, I am the same. First, one problem is that there are many errors in the detection model, and the performance of the model of the example is poor, and the detected box will appear away from the object it should detect. That is to say, the distance between the detection box and the object does not match. According to my personal considerations, I think that the network does not implement the efficient strategy proposed by the author. Since the author of the paper does not open source its code, many parameters are unknown.
Another problem is that the model is very slow to detect, even if I use tensorflow-gpu, as you have tested, the accuracy is not high, and the efficient detection strategy is not achieved. I think this is due to the wrong parameters of the network model. .
I also found this problem when I tested it. I later changed the algorithm for detecting the network when the network was not feasible in the actual scenario.
Thanks.

**Comment by scarlettliu644** [comment]
Thanks for your sharing. Cheers!
